### PR TITLE
avoid ordered arrays

### PR DESCRIPTION
**Pull request by leileiwangx**

### CODE REVIEW

**Comment by necusjz** [comment]
This repository intends to provide the most naïve solutions, it could be a follow-up question.

File: algorithms/quick_sort.py [hunk 1]
```diff
@@ -1,5 +1,10 @@
+from random import randint
+
```
Comment (necusjz) [inline algorithms/quick_sort.py]:
```suggestion
```
Built-in library doesn't need to import.

File: algorithms/quick_sort.py [hunk 2]
```diff
@@ -1,5 +1,10 @@
...
 def quick_sort(nums):
     def partition(l, r):
+        # avoid ordered arrays
+        rand_idx = randint(l, r)
+        nums[r], nums[rand_idx] = nums[rand_idx], nums[r]
```
Comment (necusjz) [inline algorithms/quick_sort.py]:
```suggestion
        idx = random.randint(l, r)
        nums[r], nums[idx] = nums[idx], nums[r]
```
More elegant.

**Review by leileiwangx — COMMENTED** [review]
LGTM

**Review by necusjz — APPROVED** [review]

**Pull request merged**